<launch>

  <arg name="custom_config" default="" />

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />

  <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- set to false if the nodelet should run under an external nodelet manager -->
  <arg name="standalone" default="true" />
  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="manager" default="$(arg UAV_NAME)_coverage_planning_manager" />
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>

  <!-- Namespace - all topics and parameters will be remapped using this namespace as a prefix (eg.: "/vision_example/image_in" to "/uav1/vision_example/image_in") -->
  <group ns="$(arg UAV_NAME)">

    <!-- TrajectoryGenerator nodelet -->
    <node pkg="nodelet" type="nodelet" name="coverage_planning_boundry_detection" args="$(arg nodelet) ns_boundry_detection/boundry_detection_node_class $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">


      <!-- ROS parameters config file -->
      <!-- <param name= "coverage_planning_area" value="$(find coverage_planning)/coverage_planning_area.txt"/> -->
      <param name= "uav_name" value="$(arg UAV_NAME)"/>

      Subscribers (note that the topic names will also be remapped using the namespace as a prefix)
      <remap from="~path_to_follow" to="coverage_planning_wrapper/path_to_follow" />
      <remap from="~update_map_service" to="coverage_planning_boustrophedon/update_map_service" />
      <remap from="~boundry_topic" to="coverage_planning_image_publisher/boundry_topic" />
      <remap from="~camera_info" to="bluefox_optflow/camera_info" />
      <remap from="~uav_state" to="control_manager/uav_state" />


      <!-- Publishers (note that the topic names will also be remapped using the namespace as a prefix) -->

    </node>

  </group>

</launch>
